android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -2,24 +2,30 @@ package coop.polypoly.polypod.polyOut
 
 import android.content.Context
 import coop.polypoly.polypod.Preferences
+import coop.polypoly.polypod.polyNav.ZipTools
 import java.io.File
 import java.nio.ByteBuffer
-import java.util.zip.ZipFile
 
 open class PolyOut(
     val context: Context
 ) {
     private val fsPrefix = "polypod://"
+
     open suspend fun readFile(path: String): ByteArray {
         if (path == "") {
             throw Error("Empty path in PolyOut.readFile")
         }
         val fs = Preferences.getFileSystem(context)
-        val entryPathStart = path.indexOf('/', fsPrefix.length)
-        val zipId = path.substring(0, entryPathStart)
-        val entryPath = path.substring(entryPathStart + 1)
-        val zip = ZipFile(File(fs[zipId]))
-        return zip.getInputStream(zip.getEntry(entryPath)).readBytes()
+
+        val filePath = context.filesDir.absolutePath + "/" + path.replace(
+            "://", ":/"
```

Comment:
What does this do? Is it because some kind of error in the specification of the path?

Reply:
I have removed this. Wasn't particularly proud of this code.